pkg/util/goroutinemap/goroutinemap.go

```diff
@@ -59,68 +59,71 @@ type GoRoutineMap interface {
 	Wait()
 }
 
-// NewGoRoutineMap returns a new instance of GoRoutineMap.
-func NewGoRoutineMap(exponentialBackOffOnError bool) GoRoutineMap {
+// NewGoroutineMap returns a new instance of GoroutineMap.
+func NewGoroutineMap(exponentialBackOffOnError bool) GoroutineMap {
 	g := &goRoutineMap{
 		operations:                make(map[string]operation),
 		exponentialBackOffOnError: exponentialBackOffOnError,
+		lock: &sync.Mutex{},
 	}
-	g.cond = sync.NewCond(g)
+
+	g.cond = sync.NewCond(g.lock)
 	return g
 }
 
 type goRoutineMap struct {
 	operations                map[string]operation
 	exponentialBackOffOnError bool
 	cond                      *sync.Cond
-	sync.Mutex
+	lock                      *sync.Mutex
```

Comment:
To address clayton's earlier comment:

> nit: we prefer lock sync.Mutex because it prevents Mutex from being public to the goRoutineMap, unless you have a need for that.
I would prefer getting access to Lock being impossible from outside this object without a good justification.